kiwi/volume_manager/lvm.py

```diff
@@ -397,18 +390,18 @@ def _is_swap_volume(self, volume_name):
             if volume_name == volume.name and volume.label == 'SWAP':
                 return True
 
-    def __del__(self):
+    def __exit__(self, exc_type, exc_value, traceback):
         if self.volume_group:
-            log.info('Cleaning up %s instance', type(self).__name__)
-            if self.umount_volumes():
-                try:
-                    Command.run(
-                        ['vgchange'] + self.lvm_tool_options + [
-                            '-an', self.volume_group
-                        ]
-                    )
-                except Exception:
-                    log.warning(
-                        'volume group %s still busy', self.volume_group
+            try:
+                self.umount_volumes()
+                Command.run(
+                    ['vgchange'] + self.lvm_tool_options + [
+                        '-an', self.volume_group
+                    ]
+                )
+            except Exception as issue:
```

Comment:
This should only catch exceptions that `Command.run` can raise, i.e. `KiwiCommandError`. Otherwise you could mask real bugs (e.g. accessing the wrong attribute and catching `AttributeError`)
```suggestion
            except KiwiCommandError as issue:
```

Reply:
done